hal/common/critical.c

```diff
@@ -304,9 +294,21 @@ uint32_t core_util_atomic_decr_u32(uint32_t *valuePtr, uint32_t delta)
     return newValue;
 }
 
-void *core_util_atomic_decr_ptr(void **valuePtr, ptrdiff_t delta) {
-    return core_util_atomic_decr((uintptr_t)valuePtr, (uintptr_t)delta);
+#endif
+
+
+bool core_util_atomic_cas_ptr(void **ptr, void **expectedCurrentValue, void *desiredValue) {
+    return (void*)core_util_atomic_cas_u32(
```

Comment:
Ah, good catch, didn't notice the return type. That's what I get for not paying enough attention.